### PR TITLE
add new sources

### PR DESCRIPTION
**Pull request by qairjar**

### CODE REVIEW

File: cassandra/client.go [hunk 1]
```diff
@@ -1 +1,25 @@
...
+
+	scyllaConf.Host = u.Host
+	scyllaConf.Keyspace = u.Query().Get("keyspace")
+	scyllaConf.User = u.User.Username()
+	scyllaConf.Type = u.Query().Get("type")
+	scyllaConf.Consistency = u.Query().Get("consistency")
+	scyllaConf.EnableAuth, _ = strconv.ParseBool(u.Query().Get("enableAuth"))
+	scyllaConf.Pass, _ = u.User.Password()
+	scyllaConf.TimeoutValid, _ = time.ParseDuration(u.Query().Get("timeoutValid"))
+	scyllaConf.ConnectTimeout, _ = time.ParseDuration(u.Query().Get("connectTimeout"))
```
Comment (artyomturkin) [inline cassandra/client.go]: Better to set default value, if param does not exist

File: kafka/sarama.go [hunk 2]
```diff
@@ -6,16 +6,11 @@ import (
...
 )
 
 // NewSaramaConfig build new sarama config from URL string.
-func NewSaramaConfig(connectionURL string) ([]string, *sarama.Config, error) {
-	u, err := url.Parse(connectionURL)
-	if err != nil {
-		return nil, nil, err
-	}
-
+func NewSaramaConfig(u *url.URL) ([]string, *sarama.Config, error) {
```
Comment (artyomturkin) [inline kafka/sarama.go]: Do not change interface, it'll break other packages

**Review by artyomturkin — CHANGES_REQUESTED** [review]

File: cassandra/client.go (hunk 1 above)
Comment (qairjar) [reply cassandra/client.go]: ok,  I will fix that

File: kafka/sarama.go (hunk 2 above)
Comment (qairjar) [reply kafka/sarama.go]: > Do not change interface, it'll break other packages

ok

File: cassandra/client.go [hunk 3]
```diff
@@ -11,15 +11,56 @@ import (
 func NewScyllaConfig(u *url.URL) (*sql.DB, error) {
 	scyllaConf := scyllaplugin.SQLConfig{}
 
+	keyspace := "test"
```
Comment (artyomturkin) [inline cassandra/client.go]: Let's make it required. If not provided throw an error

File: cassandra/client.go [hunk 4]
```diff
@@ -11,15 +11,56 @@ import (
 func NewScyllaConfig(u *url.URL) (*sql.DB, error) {
```
Comment (artyomturkin) [inline cassandra/client.go]: Use string as input parameter, to keep interfaces consistent across all connections

File: watermill/watermill.go [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+		if err != nil {
+			return nil, err
+		}
+	case "elastic":
+		elasticPub := &elasticplugin.Publisher{ElasticURL: u.Host}
+		pub, err = elasticPub.NewPublisher(nil, logger)
+		if err != nil {
+			return nil, err
+		}
+	default:
```
Comment (artyomturkin) [inline watermill/watermill.go]: Would be better to use explicit mapping `kafka` and `kafkas`

File: watermill/watermill.go [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+		brokers, conf, err := kafka2.NewSaramaConfig(connection)
```
Comment (artyomturkin) [inline watermill/watermill.go]: `kafka.NewWatermillPublisher` can be used directly here, so behavior can be the same

File: watermill/watermill.go [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+			Unmarshaler:           kafka.DefaultMarshaler{},
+		}
+
+		if cg := u.Query().Get("group"); cg != "" {
+			sc.ConsumerGroup = cg
+		}
+		c, _ = kafka.NewSubscriber(
+			sc,
+			logger,
+		)
```
Comment (artyomturkin) [inline watermill/watermill.go]: Same as for publisher

**Review by artyomturkin — CHANGES_REQUESTED** [review]

File: postgres/client.go [hunk 8]
```diff
@@ -0,0 +1,13 @@
+package postgres
```
Comment (artyomturkin) [inline postgres/client.go]: Let's rename to `sql`

File: postgres/client.go [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+
+import (
+	"database/sql"
+	_ "github.com/go-sql-driver/mysql"
+	_ "github.com/godror/godror"
+	_ "github.com/lib/pq"
```
Comment (artyomturkin) [inline postgres/client.go]: Let's leave importing drivers in `main.go`

File: postgres/client.go [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+	"net/url"
+)
+
+func SetConf(u *url.URL) (*sql.DB, error) {
+	return sql.Open(u.Scheme, u.String())
```
Comment (artyomturkin) [inline postgres/client.go]: Not all schemas match go drivers, we should use switch to adjust driver to schema. For example:

`oracle` -> `godor`

**Review by artyomturkin — CHANGES_REQUESTED** [review]

**Review by artyomturkin — APPROVED** [review]

**Pull request merged**